### ISSUE TITLE
Re-visit typescript parameters

### ISSUE THREAD

**Issue by wolfy1339** [octokit/graphql.js#47]
This is the error and no-one has yet been able to fix it. Maybe sometime later it will be obvious how to fix the issue
```
Type '(<T extends GraphQlQueryResponse>(query: string | Parameters, options?: Parameters | undefined) => Promise<{ [key: string]: any; } | null>) & { defaults: (newDefaults: Parameters) => graphql; endpoint: endpoint; }' is not assignable to type 'graphql'.
  Type 'Promise<{ [key: string]: any; } | null>' is not assignable to type 'Promise<GraphQlQueryResponse>'.
    Type '{ [key: string]: any; } | null' is not assignable to type 'GraphQlQueryResponse'.
      Type 'null' is not assignable to type 'GraphQlQueryResponse'.
```

**Comment by gr2m** [comment]
Can you share an example that will use the typescript parameter? Would it define the type of the result the `graphql()` method resolves with?

**Comment by wolfy1339** [comment]
Yes, it will change the return type.

I'm not sure I have a proper example, since I've never used GraphQL ever, but it works the same as in `@octokit/rest`.

Since you said in #46 that you didn't understand these, I'll link to the relevant section in the TS docs: https://www.typescriptlang.org/docs/handbook/generics.html

**Comment by gr2m** [comment]
Okay I think I got it now :)

**Comment by copperwall** [comment]
Hey @gr2m 👋 is this resolved by #99? It seems kind of similar.

**Comment by gr2m** [comment]
Yes, I think so, too 👍
